### PR TITLE
Feat/web3 interface hooks for v3

### PR DESCRIPTION
**Pull request by luckyOdisetti**
WIP 
* Update Config with `network` and `contracts` 
* Handle network switching and display network in `Connected Button`
* create `allowance` context
* create `wallet` context
* Update api routes with `network` as route params
* Handle token ratio

### CODE REVIEW

**Comment by kkennis** [comment]
Can you remove the `package-lock.json` files, let's stick to yarn?

File: packages/client/src/components/add-liquidity/add-liquidity-v3.scss [hunk 1]
```diff
@@ -0,0 +1,179 @@
...
+
+    .tab-container {
+        display: flex;
+        .tab {
+            font-size: 1.2rem;
+            text-align: center;
+            padding: 0.5rem 2rem;
+            flex-grow: 1;
+            h3 {
+                font-family: 'Barlow Condensed';
```
Comment (khcoaching) [inline packages/client/src/components/add-liquidity/add-liquidity-v3.scss]: shouldn't this be a global var

File: packages/client/src/components/pair-search.tsx [hunk 2]
```diff
@@ -1,23 +1,77 @@
...
 type SearchKeys = { [searchKey: string]: UniswapPair };
 
+const useStyles = makeStyles(() => ({
+    input: {
+        color: 'var(--facePrimary)',
+        background: 'var(--bgContainer)',
+    },
+}));
+
+// hack this a little more, TextField is still ugly af
```
Comment (khcoaching) [inline packages/client/src/components/pair-search.tsx]:
```suggestion
// TODO: hack this a little more, TextField is still ugly af
```

File: packages/client/src/components/pair-search.tsx [hunk 3]
```diff
@@ -59,6 +113,17 @@ function PairSearch({
         );
     };
```
Comment (khcoaching) [inline packages/client/src/components/pair-search.tsx]:
```suggestion
// TODO: move to util when we reuse this
```

File: packages/client/src/components/pair-search.tsx [hunk 4]
```diff
@@ -67,21 +132,61 @@ function PairSearch({
...
+            keys: ['token0.symbol', 'token1.symbol', 'token.pairReadable'],
+        }).slice(0, 50);
+
+    const renderPairWithLogo = (pair: UniswapPair) => (
+        <div className='pair-option-with-logo'>
+            <div className='pair'>
+                {resolveLogo(pair?.token0?.id)}&nbsp;&nbsp;
+                {pair?.token0?.symbol}
+            </div>
+            &nbsp;
```
Comment (khcoaching) [inline packages/client/src/components/pair-search.tsx]: lol

File: packages/client/src/containers/landing-container.tsx [hunk 5]
```diff
@@ -31,14 +40,17 @@ function LandingContainer({
...
+    const pairData = usePairDataOverview(pairId || null);
+    const balances = useBalance({
+        pairData,
+        wallet,
+    });
 
     (window as any).positionData = positionData;
     (window as any).pairId = pairId;
-    
+    (window as any).balances = balances;
```
Comment (khcoaching) [inline packages/client/src/containers/landing-container.tsx:50]: We should define a interface for the global window at some point

```
declare global {
    interface Window { MyNamespace: any; }
}

window.MyNamespace = window.MyNamespace || {};
```

File: packages/client/src/types/app-config.ts [hunk 6]
```diff
@@ -1,5 +1,25 @@
 export type Environments = 'production' | 'development' | 'test' | 'staging';
 
+export type NetworkNamespace = 'mainnet' | 'rinkeby' | 'goerli' | 'ropsten' | 'kovan';
```
Comment (khcoaching) [inline packages/client/src/types/app-config.ts]: @kkennis added a shared type to the somm types repo

File: packages/client/src/components/add-liquidity/add-liquidity-v3.scss (hunk 1 above)
Comment (luckyOdisetti) [reply packages/client/src/components/add-liquidity/add-liquidity-v3.scss]: yea we currently have 2 different font types that we use without a standard way of what goes where. so until then.

**Pull request merged**